src/UCTSearch.cpp

```diff
@@ -230,11 +232,11 @@ Move UCTSearch::think() {
     assert(m_playouts == 0);
     assert(m_nodes == 0);
 
-    // Start counting time for us
-//    m_rootstate.start_clock();
-
     // set up timing info
-    Time start;
+
+    Time.init(bh_.cur().side_to_move(), bh_.cur().game_ply());
+    m_target_time = get_search_time();
+    m_start_time = now();
```

Comment:
Probably not a big deal, but the reason SF did the time initialization so early is to make sure the overhead of starting the search didn't cause it to lose on time because it wasn't counted.  But this is cleaner, we could just include some small overhead time per move if we start seeing time losses.

Reply:
I have added a 30ms overhead in the calculations, to hopefully cover this